### ISSUE TITLE
bootstrap failed

### ISSUE THREAD

**Issue by djDuff** [CANDY-LINE/node-red-contrib-lwm2m#10]
Hi @dbaba 
I am trying to bootstrap using v2.1.0
LwM2M Client Out shows "bootstraping" then "disconnected".

Config of Client Out node is:
![image](https://user-images.githubusercontent.com/30717765/60101932-a2481680-9797-11e9-93b6-bf3ea356e557.png)

When I do the same using leshan client, it works:
`java -jar leshan-client-demo.jar -n ezhiand-nodered-dev3 -i ezhiand-nodered-dev3 -p <PWD> -b -u <my BSS URL>:5684`
in BSS logs I see attempts (successful and not) only for leshan client, and no attempts when using Node-Red. 
Note: There are no firewall limitations or any networking issues.
I tried to use Output LwM2M client logs and Dump lwm2m messages - but I didn't get any useful information from syslog. So I cant understand if it is client issue or configuration, and what level exactly is not working correctly: DTLS or COAP or UDP/IP
Could you suggest what else should I troubleshoot, please ?

**Comment by dbaba** [comment]
Could you find something related to the bootstrap process in BSS server logs?
And could you provide LwM2M server product name and version?
If the server is OSS, I can test this node with it. I tested this node with Leshan 1.0.0 M10 based server so it might include some server implementation specific behavior.

**Comment by dbaba** [comment]
Found something wrong on the bootstrap process. Let me check.

**Comment by djDuff** [comment]
@dbaba thank you!
I was using Leshan 1.0.0 M9 BSS. 
Nope, nothing in BSS logs.
Just to doublecheck, I switched to Eclipse Leshan public BSS and GW.
It is still not working in Node-red, but working in leshan-client-demo

here is my device configuration on GW (search for djduff-nodered-device):
https://leshan.eclipseprojects.io/#/security

Here is my device configuration on BSS (search for djduff-nodered-device):
https://leshan.eclipseprojects.io/bs/

```
java -jar leshan-client-demo.jar -n djduff-nodered-device -i djduff-nodered-device -p 313233343536 -b -u leshan.eclipseprojects.io:5784
2019-06-26 07:43:26,379 INFO LeshanClientDemo - Press 'w','a','s','d' to change reported Location (-71.0,-65.0).
2019-06-26 07:43:26,379 INFO LeshanClient - Starting Leshan client ...
2019-06-26 07:43:26,386 INFO LeshanClient - Leshan client[endpoint:djduff-nodered-device] started.
2019-06-26 07:43:26,387 INFO RegistrationEngine - Trying to start bootstrap session to coaps://leshan.eclipseprojects.io:5784 ...
2019-06-26 07:43:26,461 INFO CaliforniumEndpointsManager - New endpoint created for server coaps://23.97.187.154:5784 at coaps://0.0.0.0:52355
2019-06-26 07:43:27,549 INFO RegistrationEngine - Bootstrap started
2019-06-26 07:43:29,057 INFO RegistrationEngine - Bootstrap finished Bootstrap Server [uri=coaps://leshan.eclipseprojects.io:5784].
2019-06-26 07:43:29,070 INFO CaliforniumEndpointsManager - New endpoint created for server coaps://23.97.187.154:5684 at coaps://0.0.0.0:58624
2019-06-26 07:43:29,071 INFO RegistrationEngine - Trying to register to coaps://23.97.187.154:5684 ...
2019-06-26 07:43:30,063 INFO RegistrationEngine - Registered with location '/rd/t0bZ1P3ZrG'.
2019-06-26 07:43:30,063 INFO RegistrationEngine - Next registration update to coaps://23.97.187.154:5684 in 18s...
^C2019-06-26 07:43:34,083 INFO LeshanClient - Destroying Leshan client ...
2019-06-26 07:43:34,085 INFO RegistrationEngine - Trying to deregister to coaps://23.97.187.154:5684 ...
2019-06-26 07:43:34,335 INFO RegistrationEngine - De-register response DELETED null.
2019-06-26 07:43:34,336 INFO LeshanClient - Leshan client destroyed.
```

attaching node:
pwd for BSS is 313233343536
`[{"id":"40062642.8825d8","type":"lwm2m client out","z":"da23253d.516e98","name":"djduff-nodered-device","lwm2mClient":"c4062ed1.7ebd9","x":420,"y":1400,"wires":[]},{"id":"c4062ed1.7ebd9","type":"lwm2m client","z":"","disabled":false,"clientName":"djduff-nodered-device","enableDTLS":true,"clientPort":"56874","lifetimeSec":"300","reconnectSec":"300","requestBootstrap":true,"saveProvisionedConfig":false,"useIPv4":true,"serverHost":"leshan.eclipseprojects.io","serverPort":"5784","redirectLwm2mClientLog":false,"dumpLwm2mMessages":false,"hideSensitiveInfo":false,"propagateInternalEvents":false,"objects":"{\"3303\":{\"0\":{\"5700\":{\"type\":\"STRING\",\"acl\":\"R\",\"value\":\"\"}}}}"}]`

**Comment by dbaba** [comment]
Fixed. The root cause is an issue related to Create command on Node.js.
Can you please try v2.1.1?
Thank you for the report, very helpful.
Re-open this ticket if you still have any trouble.

**Comment by djDuff** [comment]
@dbaba Baba san, immediately after upgrade to 2.1.1 and restart it didn't start to work.
 I will troubleshoot and do more tests drom my side and get back to you in couple of days.

**Comment by dbaba** [comment]
Probably because the previously saved credential file prevented the node from re-bootstapping.
Can you please try to remove `.node-red/lwm2m_<your-endpoint-name>_cred.json`?
Will improve this option's usability later on.

**Comment by djDuff** [comment]
@dbaba I couldn't find this json file - it doesn't exist.
I deleted configuration, restarted node-red and created configuration for client out node again.
Once deployed, it was first in "bootstrapping" status, then changed to "register required"
![image](https://user-images.githubusercontent.com/30717765/60312799-62cf2500-9965-11e9-8a93-0f7ce6898cd7.png), but not in "registered"

Once re-deployed again, it is in "disconnected" status immediately.

**Comment by dbaba** [comment]
Can you please post LwM2M Client log output from bootstrap starting to the time when the node transitions to "register required" state?

**Comment by djDuff** [comment]
@dbaba Yes, here is syslog output with "Output LwM2M client logs"

`
#Deploy

Jun 28 10:59:56 dj-raspberry3b Node-RED[27107]: 28 Jun 10:59:56 - [info] Started flows
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: Trying to bind LWM2M Client to port 56872
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:objectId=>0
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: done:cmd=>[readInstances], resp=>[/resp:readInstances:AhBFAAABAAAA], base64=>[AhBFAAABAAAA], base64Len=>[12]
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:(lwm2m_data_new):numData=>1
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:result=>0x45
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: setup_instance_ids:objectId=>0:instanceId=0 (0/1)
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:objectId=>1
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: done:cmd=>[readInstances], resp=>[/resp:readInstances:AhBFAQABAAAA], base64=>[AhBFAQABAAAA], base64Len=>[12]
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:(lwm2m_data_new):numData=>1
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:result=>0x45
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: setup_instance_ids:objectId=>1:instanceId=0 (0/1)
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:objectId=>2
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: done:cmd=>[readInstances], resp=>[/resp:readInstances:AhBFAgABAAAA], base64=>[AhBFAgABAAAA], base64Len=>[12]
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:(lwm2m_data_new):numData=>1
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:result=>0x45
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: setup_instance_ids:objectId=>2:instanceId=0 (0/1)
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:objectId=>3
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: done:cmd=>[readInstances], resp=>[/resp:readInstances:AhBFAwABAAAA], base64=>[AhBFAwABAAAA], base64Len=>[12]
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:(lwm2m_data_new):numData=>1
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:result=>0x45
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: setup_instance_ids:objectId=>3:instanceId=0 (0/1)
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: LWM2M Client "djduff-nodered-device" started on port 56872
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: > prv_generic_read:objectId=>0, instanceId=>0, numData=>3
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 3] resourcId=>1
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read: [2 of 3] resourcId=>10
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read: [3 of 3] resourcId=>11
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAAAAAAMAAQAIAQABCgAGAgA5OQsABgIAMTA=], base64=>[AgFFAAAAAAMAAQAIAQABCgAGAgA5OQsABgIAMTA=], base64Len=>[40]
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>3
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>0, instanceId=>0, numData=>1
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 1] resourcId=>0
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAAAAAAEAAAAEJgBjb2FwczovL2xlc2hhbi5lY2xpcHNlcHJvamVjdHMuaW86NTc4NA==], base64=>[AgFFAAAAAAEAAAAEJgBjb2FwczovL2xlc2hhbi5lY2xpcHNlcHJvamVjdHMuaW86NTc4NA==], base64Len=>[72]
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>1
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>0, instanceId=>0, numData=>1
Jun 28 10:59:57 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 1] resourcId=>2
Jun 28 10:59:58 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAAAAAAEAAgAGAQAw], base64=>[AgFFAAAAAAEAAgAGAQAw], base64Len=>[20]
Jun 28 10:59:58 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>1
Jun 28 10:59:58 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 10:59:58 dj-raspberry3b Node-RED[27107]: backup_object:objectId=>0
Jun 28 10:59:58 dj-raspberry3b Node-RED[27107]: done:cmd=>[backup], resp=>[/resp:backup:AgEAAAAAAAAA], base64=>[AgEAAAAAAAAA], base64Len=>[12]
Jun 28 10:59:58 dj-raspberry3b Node-RED[27107]: backup_object:result=>0x0
Jun 28 10:59:58 dj-raspberry3b Node-RED[27107]: backup_object:objectId=>1
Jun 28 10:59:58 dj-raspberry3b Node-RED[27107]: done:cmd=>[backup], resp=>[/resp:backup:AgEAAQAAAAAA], base64=>[AgEAAQAAAAAA], base64Len=>[12]
Jun 28 10:59:58 dj-raspberry3b Node-RED[27107]: backup_object:result=>0x0


#bootstraping 


Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: 60 bytes received from [23.97.187.154]:5784
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: 120 bytes received from [23.97.187.154]:5784
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>0, instanceId=>0, numData=>1
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 1] resourcId=>3
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAAAAAAEAAAAFFQBkamR1ZmYtbm9kZXJlZC1kZXZpY2U=], base64=>[AgFFAAAAAAEAAAAFFQBkamR1ZmYtbm9kZXJlZC1kZXZpY2U=], base64Len=>[48]
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>1
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>0, instanceId=>0, numData=>1
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 1] resourcId=>5
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAAAAAAEAAAAFBgAxMjM0NTY=], base64=>[AgFFAAAAAAEAAAAFBgAxMjM0NTY=], base64Len=>[28]
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>1
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 11:00:07 dj-raspberry3b Node-RED[27107]: 67 bytes received from [23.97.187.154]:5784
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: 37 bytes received from [23.97.187.154]:5784
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: 43 bytes received from [23.97.187.154]:5784
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_delete:objectId=>0, instanceId=>0
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: 28 Jun 11:00:09 - [error] [lwm2m client:25618238.56f52e] [Delete] (^/0/0/[0-9]+$) Error Message:Unauthorized (insufficient permission)
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[delete], resp=>[/resp:delete:AgGBAAAAAAAA], base64=>[AgGBAAAAAAAA], base64Len=>[12]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_delete:result=>0x81
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: 43 bytes received from [23.97.187.154]:5784
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_delete:objectId=>1, instanceId=>0
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[delete], resp=>[/resp:delete:AgFCAQAAAAAA], base64=>[AgFCAQAAAAAA], base64Len=>[12]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_delete:result=>0x42
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: 148 bytes received from [23.97.187.154]:5784
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_create:objectId=>0, instanceId=>0, numData=>13
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[create], resp=>[/resp:create:AgFBAAAAAAAA], base64=>[AgFBAAAAAAAA], base64Len=>[12]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_create:result=>0x41
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: 145 bytes received from [23.97.187.154]:5784
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_create:objectId=>0, instanceId=>1, numData=>13
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[create], resp=>[/resp:create:AgFBAAABAAAA], base64=>[AgFBAAABAAAA], base64Len=>[12]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_create:result=>0x41
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: 64 bytes received from [23.97.187.154]:5784
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_create:objectId=>1, instanceId=>0, numData=>5
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[create], resp=>[/resp:create:AgFBAQAAAAAA], base64=>[AgFBAQAAAAAA], base64Len=>[12]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_create:result=>0x41
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: 44 bytes received from [23.97.187.154]:5784
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>0, instanceId=>1, numData=>3
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 3] resourcId=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [2 of 3] resourcId=>10
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [3 of 3] resourcId=>11
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAAABAAMAAQAFAQAACgAFAQB7CwAFAQAB], base64=>[AgFFAAABAAMAAQAFAQAACgAFAQB7CwAFAQAB], base64Len=>[36]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>3
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>1, instanceId=>0, numData=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 1] resourcId=>0
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAQAAAAEAAAAFAQB7], base64=>[AgFFAQAAAAEAAAAFAQB7], base64Len=>[20]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>1, instanceId=>0, numData=>2
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 2] resourcId=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [2 of 2] resourcId=>7
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAQAAAAIAAQAFAQAeBwAFAQBV], base64=>[AgFFAQAAAAIAAQAFAQAeBwAFAQBV], base64Len=>[28]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>2
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>0, instanceId=>1, numData=>3
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 3] resourcId=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [2 of 3] resourcId=>10
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [3 of 3] resourcId=>11
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAAABAAMAAQAFAQAACgAFAQB7CwAFAQAB], base64=>[AgFFAAABAAMAAQAFAQAACgAFAQB7CwAFAQAB], base64Len=>[36]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>3
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>1, instanceId=>0, numData=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 1] resourcId=>0
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAQAAAAEAAAAFAQB7], base64=>[AgFFAQAAAAEAAAAFAQB7], base64Len=>[20]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>1, instanceId=>0, numData=>2
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 2] resourcId=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [2 of 2] resourcId=>7
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAQAAAAIAAQAFAQAeBwAFAQBV], base64=>[AgFFAQAAAAIAAQAFAQAeBwAFAQBV], base64Len=>[28]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>2
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:objectId=>0, instanceId=>1, numData=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read: [1 of 1] resourcId=>0
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[read], resp=>[/resp:read:AgFFAAABAAEAAAAFJgBjb2FwczovL2xlc2hhbi5lY2xpcHNlcHJvamVjdHMuaW86NTY4NA==], base64=>[AgFFAAABAAEAAAAFJgBjb2FwczovL2xlc2hhbi5lY2xpcHNlcHJvamVjdHMuaW86NTY4NA==], base64Len=>[72]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:(lwm2m_data_new):numData=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read:result=>0x45
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: Connection creation failed.
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: lwm2m_step() failed: 0xA3
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: restore_object:objectId=>0
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[restore], resp=>[/resp:restore:AgEAAAAAAAAA], base64=>[AgEAAAAAAAAA], base64Len=>[12]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: restore_object:result=>0x0
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:objectId=>0
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[readInstances], resp=>[/resp:readInstances:AhBFAAABAAAA], base64=>[AhBFAAABAAAA], base64Len=>[12]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:(lwm2m_data_new):numData=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:result=>0x45
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: setup_instance_ids:objectId=>0:instanceId=0 (0/1)
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: restore_object:objectId=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[restore], resp=>[/resp:restore:AgEAAQAAAAAA], base64=>[AgEAAQAAAAAA], base64Len=>[12]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: restore_object:result=>0x0
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:objectId=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: done:cmd=>[readInstances], resp=>[/resp:readInstances:AhBFAQABAAAA], base64=>[AhBFAQABAAAA], base64Len=>[12]
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:(lwm2m_data_new):numData=>1
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: prv_generic_read_instances:result=>0x45
Jun 28 11:00:09 dj-raspberry3b Node-RED[27107]: setup_instance_ids:objectId=>1:instanceId=0 (0/1)


#register required
`

**Comment by dbaba** [comment]
Thanks. I identified the error cause. The provisioned server url is OPAQUE but wakaama client accepts only STRING, which failed to handle URL string. Will address it soon.

**Comment by dbaba** [comment]
Fixed in v2.1.2.

**Comment by djDuff** [comment]
Yes, confirmed. Working fine! Thank you @dbaba !
